addons/last-edit-tooltip/userscript.js

```diff
@@ -1,15 +1,34 @@
 export default async function ({ addon, global, console, msg }) {
-  fetch("https://api.scratch.mit.edu" + document.location.pathname)
-    .then(function (response) {
-      return response.json();
-    })
-    .then(async function (text) {
-      while (true) {
-        const element = await addon.tab.waitForElement(".share-date", { markAsSeen: true });
-        if (text.history) {
-          let dateMod = new Date(text.history.modified);
-          element.setAttribute("title", msg("modified", { date: scratchAddons.l10n.date(dateMod) }));
-        }
-      }
-    });
+  const data = await (
+    await fetch("https://api.scratch.mit.edu" + location.pathname.match(/\/projects\/[0-9]+/g)[0])
+  ).json();
+
+  while (true) {
+    const element = await addon.tab.waitForElement(".share-date", { markAsSeen: true });
+
+    if (!data.history) return;
+
+    let dateMod = scratchAddons.l10n.date(new Date(data.history.modified));
+    let dateCreated = new Date(data.history.shared);
+    let hour = dateCreated.getHours();
+    let minute = (dateCreated.getMinutes() + "").padStart(2, "0");
+    let dataTitle =
+      msg("modified", { date: dateMod }) +
+      msg("shared24", {
+        date: scratchAddons.l10n.date(dateCreated),
+        hour,
+        minute,
+      });
+    if (addon.settings.get("12hrclock"))
+      dataTitle =
+        msg("modified", { date: dateMod }) +
+        msg("shared", {
+          date: scratchAddons.l10n.date(dateCreated),
+          hour: hour % 12 || 12,
+          minute,
+          hourType: hour > 11 ? "pm" : "am",
```

Comment:
Not everyone uses am/pm, I think you should use [toLocaleTimeString](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleTimeString), passing in the language (`msg.locale`).
cc @apple502j? (not sure what `scratchAddons.l10n.date` does)

Reply:
Yeah actually I think this should be handled by the localization library (Scratch language "en" might be en-US or en-GB, one uses 12hr and other uses 24hr). Waiting for @apple502j to tell us the proper way to do this

Reply:
any updates @WorldLanguages?

Reply:
@FunctionalMetatable this can't be released in time for v1.12.0 because it adds new locales, will be merged for v1.13.0, you can unwatch this PR if you want, we'll handle any needed changes

Reply:
> @FunctionalMetatable this can't be released in time for v1.12.0 because it adds new locales, will be merged for v1.13.0, you can unwatch this PR if you want, we'll handle any needed changes

alrighty - thanks!